sql/hive/src/main/scala/org/apache/spark/sql/hive/orc/OrcFileFormat.scala

```diff
@@ -118,6 +120,11 @@ class OrcFileFormat extends FileFormat with DataSourceRegister with Serializable
     val broadcastedHadoopConf =
       sparkSession.sparkContext.broadcast(new SerializableConfiguration(hadoopConf))
 
+    val enableVectorizedReader: Boolean =
+      sparkSession.sessionState.conf.orcVectorizedReaderEnabled &&
+      dataSchema.forall(f => f.dataType.isInstanceOf[AtomicType] &&
```

Comment:
This is similar with ParquetFileFormat does. We might not add new `AtomicType` frequently and current `AtomicType` should be relatively stable. If we do, it should be easily tested out the data type is not supported in reader codes.